packages/ai/src/AnalysisResultsContainer.tsx

```diff
@@ -8,16 +8,43 @@ import {
   useScrollToBottom,
   useScrollToBottomButton,
 } from './hooks/use-scroll-to-bottom';
+import {TOOLS} from './analysis';
 
 export const AnalysisResultsContainer: React.FC = () => {
   const isRunningAnalysis = useStoreWithAi((s) => s.ai.isRunningAnalysis);
   const currentSession = useStoreWithAi((s) => s.ai.getCurrentSession());
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  // TODO: need to wait the datasets to be loaded before rendering the persisted analysis results
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      setIsLoading(false);
+    }, 2000);
+
+    return () => clearTimeout(timer);
+  }, []);
```

Comment:
@ilyabo Is there anyway to know the datasets have already been loaded? I want to render the AnalysisResults after datasets been loaded into duckdb. Thanks!

Reply:
currently you can do this to check whether the data has been loaded (on initialization):
```
const isDataAvailable = useProjectStore(state => state.project.isDataAvailable)
```
or for a particular table:
```
  const isTableReady = useProjectStore((state) =>
    Boolean(state.project.tables.find((t) => t.tableName === 'mammals')),
  );
  ```